index.js

```diff
@@ -62,7 +120,192 @@ function createTaskCreatedAt(taskCreatedAt) {
 function createTaskCheckBox() {
   return `<input
             type="checkbox"
-            class="table-box-check-box"
             name="checkbox-done"
           /> `;
 }
+
+function sortTasks(typeOfSort) {
+  if (typeOfSort == "a-to-z") {
+    sortTasksAtoZ();
+  } else {
+    unsortTasks();
+  }
+}
+
+function sortTasksAtoZ() {
+  const collator = new Intl.Collator("en", {
+    numeric: true,
+  });
+
+  tasks = tasks.sort((randomTask1, randomTask2) =>
+    collator.compare(
+      randomTask1.title.toUpperCase(),
+      randomTask2.title.toUpperCase()
+    )
+  );
+
+  renderTasks();
+}
+
+function unsortTasks() {
+  tasks.sort((randomTask1, randomTask2) => {
+    const randomTask1ID = randomTask1.id;
+    const randomTask2ID = randomTask2.id;
+    let comparisonResult = 0;
+    if (randomTask1ID > randomTask2ID) {
+      comparisonResult = 1;
+    } else if (randomTask1ID < randomTask2ID) {
+      comparisonResult = -1;
+    }
+    return comparisonResult;
+  });
+
+  renderTasks();
+}
+
+function searchTitle(text) {
+  if (text != null || "") {
+    tasks = tasks.filter((task) => task.title.includes(text));
+    renderTasks();
+    tasks = allTasks;
+  } else {
+    renderTasks();
+  }
+}
+
+function filterTasksByTime(time) {
```

Comment:
This function needs some refactor based on the comments below.

Reply:
got it